### ISSUE TITLE
Many break-outside-of-loop false negatives

### ISSUE THREAD

**Issue by jplatte** [rust-lang/rust-analyzer#14503]
It looks like #14486, while it fixed type-mismatch errors, introduced a new kind of error on `?`s <del>inside `async fn`s</del>` inside closures or async blocks that occur within an `async fn`:

![Screenshot_2023-04-06_091317](https://user-images.githubusercontent.com/951129/230302074-cefb6e48-78c9-4f22-bab3-d6d8961b7cd7.png)

**rust-analyzer version**: 0.4.1465-standalone

**rustc version**: 1.69.0-beta.1 (b955c8271 2023-03-06)

**relevant settings**: -

cc @HKalbasi

**Comment by Veykril** [comment]
Does this happen for all your async functions? Are attributes involved? I can't reproduce htis with a simple
```rs

async fn f() {
    None::<()>.await?;
}
```

**Comment by jplatte** [comment]
Aha, yes attributes are involved in many places. Though I've also found some closures without attributes where this happens. It seems like both closures inside `async fn` and `async` blocks inside `async fn` (often from `#[tracing::instrument]`) are triggering this.

![Screenshot_2023-04-06_135440](https://user-images.githubusercontent.com/951129/230371455-1873f66d-c290-4b94-affc-08484dfb5d9a.png)

(first error disappears when removing `instrument`, all errors disappear when removing `async` before `fn`)

![Screenshot_2023-04-06_135710](https://user-images.githubusercontent.com/951129/230372873-fb549c93-8f2e-4b30-8ea4-23ef13f451df.png)

(error disappears when removing `async` before `fn`)
